sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -4129,6 +4129,36 @@ class Dataset[T] private[sql](
     new DataFrameWriterV2[T](table, this)
   }
 
+  /**
+   * Create a [[DataFrameWriterV2]] for MergeInto action.
+   *
+   * Scala Examples:
+   * {{{
+   *   spark.table("source")
+   *     .mergeInto("target")
+   *     .on($"source.id" === $"target.id")
+   *     .whenMatched($"salary" === 100)
+   *     .delete()
+   *     .whenNotMatched()
+   *     .insertAll()
+   *     .whenNotMatchedBySource($"salary" === 100)
+   *     .update(Map(
+   *       "salary" -> lit(200)
+   *     ))
+   *     .merge()
+   * }}}
+   *
+   * @since 4.0.0
+   */
+  def mergeInto(table: String): DataFrameWriterV2[T] = {
+    if (isStreaming) {
```

Comment:
the implementation looks identical to `writeTo`?

Reply:
After introducing a new `MergeIntoWriter`, it became `def mergeInto(table: String): MergeIntoWriter[T]`